### PR TITLE
fix bug where buddy checks didn't work when deploy groups feature enabled

### PR DESCRIPTION
**Pull request by jonmoter**
since we redefined the Stage#production? method, that logic needs to be run in ruby instead of via a SQL query.
also refactored code to consolidate the places we check the DEPLOY_GROUP_FEATURE env var.

/cc @zendesk/samson, @zendesk/runway 

### Risks
 - Logic fix around buddy checks, when a past approval affects a new deploy

### CODE REVIEW

File: app/models/deploy_service.rb [hunk 1]
```diff
@@ -32,16 +32,13 @@ def auto_confirm?(stage)
   end
 
   def latest_approved_deploy(reference, project)
-    deploy = nil
-    Deploy.where(reference: reference).where(' buddy_id is NOT NULL AND started_at > ?', BuddyCheck.period.hours.ago)
+    Deploy.where(reference: reference).where('buddy_id is NOT NULL AND started_at > ?', BuddyCheck.period.hours.ago)
       .includes(:stage)
-      .where(stages: {project_id: project, production: true})
+      .where(stages: {project_id: project})
       .each do |d|
```
Comment (grosser) [inline app/models/deploy_service.rb]: `.detect` ?

File: app/views/projects/_stage.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
       <%= link_to stage.name, [project, stage] %>
       <%= stage_lock_icon stage %>
     </td>
-    <% if ENV['DEPLOY_GROUP_FEATURE'] %>
+    <% if DeployGroup.enabled? %>
```
Comment (grosser) [inline app/views/projects/_stage.html.erb]: :heart: nice cleanup

**Comment by grosser** [comment]
:+1:

**Comment by grosser** [comment]
and the detect ?

File: app/models/deploy_service.rb (hunk 1 above)
Comment (jonmoter) [inline app/models/deploy_service.rb]: Ah, I missed that comment.  Yeah, `detect` would be better here.  I'll create another PR to clean up this class a little bit.

**Pull request merged**